File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file215.f90
subroutine kernel_floyd_warshall(n, path)
implicit none

DATA_TYPE, dimension(n,n) :: path
integer :: n
integer :: i, j, k

!$pragma scop
do k=1, _PB_N
do i=1, _PB_N
do j=1, _PB_N
if( path(j, i) .GE. path(k, i) + path(j, k) ) then
path(j, i) = path(k, i) + path(j, k)
end if
end do
end do
end do
!$pragma endscop
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file215.cpp
void kernel_floyd_warshall(int n, DATA_TYPE path[][n])
{
 for(int k = 0; k < _PB_N; k++)
 for(int i = 0; i < _PB_N; i++)
 for(int j = 0; j < _PB_N; j++)
 if(path[j][i] >= path[k][i] + path[j][k])
 path[j][i] = path[k][i] + path[j][k];
}
